File: programmers/javascript/level3/단속카메라.js
const routes = [
  [-20, 15],
  [-14, -5],
  [-18, -13],
  [-5, -3],
];

const solution = (routes) => {
  let answer = 0;
  routes.sort((a, b) => {
    return a[1] - b[1];
  });
  let camera = -30001;
  for (let i = 0; i < routes.length; i++) {
    if (camera < routes[i][0]) {
      answer++;
      camera = routes[i][1];
    }
  }
  return answer;
};

console.log(solution(routes));

File: python/programmers/level3/단속카메라.py
def solution(routes):
    answer = 0
    routes.sort(key=lambda x: x[1])

    camera = -30001
    for x in routes:
        start = x[0]
        end = x[1]
        if start <= camera <= end:
            continue
        else:
            answer += 1
            camera = end
    return answer


print(solution([[-20, 15], [-14, -5], [-18, -13], [-5, -3]]))
